### ISSUE TITLE
Feature: `partitionByComment` support `line` and `block`

### ISSUE THREAD

**Issue by ntnyq** [azat-io/eslint-plugin-perfectionist#432]
### What rule do you want to change?

any rule support `partitionByComment`

### Describe the problem

Mostly is issue mixed with `jsdoc`.

Comments are used to explain code but also causes sorting behavior unexpected.

### Code example

With `sort-objects` config to `{ type: "alphabetical", order: "asc", partitionByComment: true, }`

```ts
export const foobar = {
  foo: 'foo',

  /**
   * This is bar
   */
  bar: 'bar',
}
```

If `partitionByComment: 'line'` is supported. It can be fixed automaticly.

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by OlivierZal** [comment]
Hi @ntnyq, you can also fill `partitionByComment` with a pattern or list of patterns instead of `true`, could it solve your issue?

**Comment by ntnyq** [comment]
> Hi @ntnyq,
> 
> I think it's a good use case.
> 
> You can also fill `partitionByComment` with a pattern or list of patterns instead of `true`.

Didn't notice it supports string pattern before. However, as I tried, it only match the comment content but not comment type.

**Comment by OlivierZal** [comment]
Yes, indeed it proposes just a workaround.

Did you try a regex which excludes multiline comments, e.g. `partitionByComment: "^[^\n]*$"`

**Comment by ntnyq** [comment]
> Yes, indeed it proposes just a workaround.
> 
> Did you try a regex which excludes multiline comments, e.g. `partitionByComment: "^[^\n]*$"`

Yes, `partitionByComment: "^[^\n]*$"` works fine.

**Comment by ntnyq** [comment]
Still, we can enhance the `partitionByComment` schema to allow config like:

```ts
type PartitionByComment = 
	| boolean 
	| string 
	| string[] 
	| { 
		type?: 'line' | 'block', 
		pattern?: string | string[] 
	}
```

**Comment by hugop95** [comment]
> Still, we can enhance the `partitionByComment` schema to allow config like:
> 
> ```ts
> type PartitionByComment = 
> 	| boolean 
> 	| string 
> 	| string[] 
> 	| { 
> 		type?: 'line' | 'block', 
> 		pattern?: string | string[] 
> 	}
> ```

I think that

```ts
type PartitionByComment = 
	| boolean 
	| string 
	| string[] 
	| { 
		line?: boolean | string | string[] 
		block?: boolean | string | string[]
	}
```

**Comment by ntnyq** [comment]
Agreed. It's better.

**Comment by azat-io** [comment]
Added in dd71f1108634c737f3a1f20486b02605ee61f9f3.
Released in [v4.5.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.5.0).
